consumer/consumer.go

```diff
@@ -156,9 +156,10 @@ func (c consumerGroupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, cla
 	// and it's not possible to remove a topic from the handler list.
 	h := c.consumer.handlers.get(claim.Topic())
 	for msg := range claim.Messages() {
+		mark := true
 		if err := h.handleMessage(ctx, msg, claim); err != nil {
 			if c.consumer.config.Discarded != nil {
-				c.consumer.config.Discarded(msg, err)
+				mark = c.consumer.config.Discarded(ctx, msg, err)
```

Comment:
This means, whenever an error occurred, if a discard configuration, use it to determine if the message should be marked, and rely on it to dump the error.
If no discard configuration exists this imply there is no logs, is it expected?

Reply:
`felice` does not have log facility on purpose. So yeah this is expected.